federation/server_room.go

```diff
@@ -140,6 +147,21 @@ func (r *ServerRoom) WaiterForEvent(eventID string) *helpers.Waiter {
 	return w
 }
 
+func (r *ServerRoom) WaiterForMembershipEvent(userID string) *helpers.Waiter {
```

Comment:
We could make this a general `WaiterForStateEvent` thing 🤷 

Might make it more appetizing to write engineered tests